src/liboslexec/runtimeoptimize.cpp

```diff
@@ -1827,7 +1827,7 @@ RuntimeOptimizer::eliminate_middleman ()
                             dsym = downinst->mastersymbol(c.dst.param);
                         const Symbol *usym = upinst->symbol(upstream_symbol);
                         if (! usym)
-                            usym = downinst->mastersymbol(upstream_symbol);
+                            usym = upinst->mastersymbol(upstream_symbol);
```

Comment:
Yes. Completely asymptomatic, but I fixed it when I was in the vicinity and noticed it.